drv/lpc55-sprot-server/src/main.rs

```diff
@@ -182,31 +186,38 @@ fn main() -> ! {
                 // as instructed by the SP. We do that and then proceed to wait
                 // for the next request.
                 ringbuf_entry!(Trace::Flush);
+                let _ = io.cleanup_after_request();
                 io.deassert_rot_irq();
                 continue;
             }
             Err(IoError::Flow) => {
                 ringbuf_entry!(Trace::FlowError);
                 handler.flow_error(tx_buf)
             }
+            Err(IoError::Desynchronized) => {
+                ringbuf_entry!(Trace::Desynchronized);
+                handler.desynchronized_error(tx_buf)
+            }
         };
 
+        if io.cleanup_after_request().is_err() {
+            // Reply with a desync error, not whatever the response was
+            rsp_len = handler.desynchronized_error(tx_buf);
+        }
+
         ringbuf_entry!(Trace::Stats(io.stats));
         io.reply(&tx_buf[..rsp_len]);
     }
 }
 
+struct DesynchronizedError;
 impl Io {
     // Wait for chip select to be asserted
     // Assert ROT_IRQ if this is a reply
```

Comment:
Remove this comment if we're not doing this part anymore

Reply:
Good catch. Begone!